### PR TITLE
Update a few libraries

### PR DESCRIPTION
**Pull request by joshua-cogliati-inl**
--------
Pull Request Description
--------
##### What issue does this change request address? 
#1972 

##### What are the significant changes in functionality due to this change request?
Updates scikit-learn, statsmodels and ray
Removes using some deprecated python features.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by PaulTalbot-INL** [comment]
Hm, some of the regolds look like notable changes to me. Does anyone know how "analytic" or not those gold files were?

**Comment by joshua-cogliati-inl** [comment]
> Hm, some of the regolds look like notable changes to me. Does anyone know how "analytic" or not those gold files were?

The clustering/datamining ones probably are not very analytic.

**Comment by joshua-cogliati-inl** [comment]
Tests affected:

- tests/framework/PostProcessors/DataMiningPostProcessor/DimensionalityReduction/SpectralEmbedding
- tests/framework/PostProcessors/LimitSurface/testLimitSurfaceIntegralPPWithBoundingError
- tests/framework/PostProcessors/TemporalDataMiningPostProcessor/Clustering/MiniBatchKMeans

**Comment by joshua-cogliati-inl** [comment]
Comments on differences:

The tests/framework/PostProcessors/DataMiningPostProcessor/DimensionalityReduction/SpectralEmbedding is mostly due to a few points being fairly different.
The biggest difference is the 10th number:

4.9 | 3.1 | 1.5 | 0.1 | 0 | 10 | 0.0283713757382 | -0.016950871197
-- | -- | -- | -- | -- | -- | -- | --

versus:

4.9 | 3.1 | 1.5 | 0.1 | 0 | 10 | 0.0288232499833 | -0.015954529555
-- | -- | -- | -- | -- | -- | -- | --


The tests/framework/PostProcessors/LimitSurface/testLimitSurfaceIntegralPPWithBoundingError differences are solely due to the EventProbability_err being different.  I am wondering if this is being calculated by subtracting two numbers that very close.
Original Windows:
0.01365
Original Other:
0.01371875
New Windows:
0.0138
New Linux:
0.01388125
New Mac:
0.013725

The tests/framework/PostProcessors/TemporalDataMiningPostProcessor/Clustering/MiniBatchKMeans look quantitatively similar:
centroid_0

@wangcj05  or @PaulTalbot-INL  any more questions?

**Review by wangcj05 — COMMENTED** [review]
Changes are good, I just have one question about ray. @joshua-cogliati-inl

File: dependencies.xml [hunk 1]
```diff
@@ -60,7 +60,7 @@ Note all install methods after "main" take
     <nomkl os='linux' skip_check='True'/>
     <numexpr os='linux'/>
     <cmake skip_check='True' optional='True'/>
-    <ray source="pip" pip_extra="[default]">1.12</ray>
+    <ray source="pip" pip_extra="[default]">1.13</ray>
```
[the remark this answers is not in the document]
Comment (joshua-cogliati-inl) [reply dependencies.xml:63]: I will make another branch and see what happens if we switch to ray 2.0

[the remark this answers is not in the document]
Comment (joshua-cogliati-inl) [reply dependencies.xml]: It worked, so I switched this pull request to use ray 2.0

[the remark this answers is not in the document]
Comment (joshua-cogliati-inl) [reply dependencies.xml]: Hm, ray 2.0 seems to be causing random failures in InternalParallel tests.

**Review by wangcj05 — APPROVED** [review]
changes are good.

**Comment by wangcj05** [comment]
Checklist is good. Test are green. PR can be merged.

**Comment by wangcj05** [comment]
@joshua-cogliati-inl Could you send out an email to user group since there are libraries update, and the users need to update their libraries also.

**Pull request merged**